### PR TITLE
feat(evidence): enforce provenance trust in briefs, context, and promotion

### LINKED ISSUE #587: evidence: enforce provenance trust in briefs, context, and promotion
Parent: #505
Proposal: #581
Depends on: #585 and #586

Land the final trust enforcement and auditable upgrade paths after every producer and public read surface carries the shared envelope.

## Scope

- Enforce `brigade.trust-policy.v1` in evidence briefs, Aboyeur prompt injection, context packs, work-import promotion, citations, and handoff promotion.
- Exclude unknown and quarantined content from default brief, context, cite, and promote paths.
- Wrap and cap untrusted brief content at 2 items and 50 percent of brief bytes.
- Resolve pending injection scans synchronously before any content-emitting consumer or omit the body on hit/error/unavailability.
- Add explicit operator review by item ref plus expected envelope content hash.
- Extend `brigade receipts verify` to upgrade indexed verify-receipt items only after complete receipt v2 patch identity and exact retained patch validation.
- Append work/research transition JSONL events and MiseLedger event rows.

## Acceptance

- [ ] Unknown and quarantined content cannot enter default briefs, context packs, citations, or promotion.
- [ ] Untrusted content is visibly wrapped and both caps are tested at boundaries.
- [ ] Pending, flagged, or error injection state emits metadata only and cannot leak content.
- [ ] `brigade evidence trust review` requires an exact current envelope digest and appends one reviewed transition event.
- [ ] Receipt indexing stays untrusted; verify requires `baseline_commit`, `tree_fingerprint`, matching `changes_patch_sha256`, and unchanged envelope content.
- [ ] Successful verify appends one verified event; duplicate verify is an idempotent no-op.
- [ ] Envelope verified status does not bypass #502 `subject_binding`, `check_role`, patch identity, or #474 failure taxonomy requirements.
- [ ] Full `./scripts/verify` passes through a Brigade receipt.

Size: M. This is the last implementation slice.

### PR DESCRIPTION
**Pull request by solomonneas**
## What and why

Land the final provenance trust gate after every producer and public read surface carries the shared envelope. Default briefs, context packs, citations, and promotion now enforce `brigade.trust-policy.v1`, with an auditable operator review path and a receipts-verify upgrade that cannot launder trust.

Fixes #587
Parent: #505
Proposal: #581
Depends on: #585 and #586

## Type of change

- [x] Surface change (new harness, depth, include, or breaking template/ingester change) — opened an issue first per CONTRIBUTING.md

## Behavior

- Unknown and quarantined content cannot enter default briefs, context packs, citations, or promotion.
- Untrusted brief content is wrapped and capped at 2 items and 50 percent of brief bytes (both boundaries tested).
- Pending injection is resolved synchronously. Pending, flagged, or error state emits metadata only and cannot leak a body.
- `brigade evidence trust review <item-ref> --content-hash <digest>` requires the exact current envelope digest and appends one reviewed transition event. Duplicate review is an idempotent no-op.
- Receipt indexing stays untrusted. `brigade receipts verify` upgrades indexed verify-receipt items only after complete receipt v2 `baseline_commit`, `tree_fingerprint`, matching `changes_patch_sha256`, and unchanged envelope content.
- Envelope `verified` does not bypass #502 `subject_binding`, `check_role`, patch identity, or the #474 failure taxonomy.
- Work/research `brigade.provenance-event.v1` JSONL events and MiseLedger `trust review` rows are appended on successful transitions.

Out of scope: #941 (brief line hygiene) and #952 (residual authority leak).

## Checklist

- [x] Targeted pytest for the touched modules passes locally (544 passed)
- [x] Added or updated probe-style tests (forged/laundered trust claims, wrap/cap boundaries, injection metadata-only, review digest, verify idempotency)
- [x] Updated the `Unreleased` section of `CHANGELOG.md` for any user-visible effect (entries describe effects, not commit subjects)
- [x] No personal details, hostnames, IPs, account names, tokens, or unredacted absolute paths in code, templates, tests, or this PR (the `content-guard` CI job will fail otherwise)
- [x] No new runtime dependencies (Brigade is zero-runtime-dep on purpose)
- [x] Conventional commit messages

## Verification

```text
python -m pytest -q tests/test_provenance_envelope.py tests/test_aboyeur.py tests/test_context_cmd.py tests/test_receipts_cmd.py tests/test_evidence_cmd.py tests/test_work_cmd_imports.py tests/test_research_provenance.py
544 passed in 90.69s

cd engines/evidence-ledger && go test ./internal/ingest/ -count=1 -run 'TestReviewTrustLabel'
ok
```

Full `./scripts/verify` is the CI / Brigade-receipt gate and was not run in this session.

[Targeted pytest results](https://cursor.com/agents/bc-96bd632d-adf6-46b2-ac1d-81bd006b4bfc/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Ftargeted_pytest_trust_enforcement.txt)

<sub>To show artifacts inline, <a href="https://cursor.com/dashboard/cloud-agents#my-pull-requests">enable</a> in settings.</sub>

<div><a href="https://cursor.com/agents/bc-96bd632d-adf6-46b2-ac1d-81bd006b4bfc?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-96bd632d-adf6-46b2-ac1d-81bd006b4bfc&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by solomonneas** [comment]
VERDICT: NEEDS WORK

Independent review of #965 (head `2c336b20`) against #587. Scope match is good — all 22 touched files belong to the trust-enforcement slice, nothing strays into #941 or #952, and no secrets (fixtures use `probe`/`reader` placeholders and fake digests). Test quality on the new paths is genuinely probe-style (forged digests, boundary caps, idempotent re-review). But CI is red on three jobs and one probe confirms the exact normalization bypass class found on #960: a case/format-variant injection status leaks the full item body into default briefs.

## How this was reviewed

Applied the PR head in a scratch clone (`/tmp/pr965-scratch`, deleted after) and executed, not just read:

- Targeted pytest on the 7 touched modules: **544 passed** (matches the PR body claim).
- `mypy src/brigade/trust_gate.py`: **3 errors**, identical to the failing CI `lint` job.
- `BRIGADE_EXTRAS=1 brigade roadmap commands --write`: produces a diff — `docs/command-inventory.md` is stale, identical to the failing CI `repo-metadata` job.
- Go engine: `go build`, `go vet`, `go test ./internal/ingest/ ./internal/app/` all pass; built a real `miseledger` binary and drove `trust review` CLI probes against a temp HOME.
- Python probes against `trust_gate` / `evidence_brief` / `work_cmd` for laundering, fail-closed, idempotency, and normalization (scripts deleted with the scratch clone).
- Reproduced both failing CI `test (3.11)` tests locally on the PR head.

## Blocking findings

### 1. Case/format-variant injection status leaks the body into default briefs (the #960 bypass class)

`provenance.injection_status` (src/brigade/provenance.py:473) returns the raw status string with no normalization, and `trust_gate.admit_consumer` only treats the exact lowercase values `pending`/`flagged`/`error` as blocking (src/brigade/trust_gate.py:162, :218). The wrapped-brief branch (src/brigade/trust_gate.py:239) requires only `allows(label, "brief_wrapped")` — it never requires `status == "clean"`. Observed on the PR head:

```
injection status 'flagged'   body_mode=metadata  BODY_LEAK=False   (baseline, correct)
injection status 'Flagged'   body_mode=wrapped   BODY_LEAK=True
injection status 'FLAGGED'   body_mode=wrapped   BODY_LEAK=True
injection status 'flagged '  body_mode=wrapped   BODY_LEAK=True
injection status 'Error'     body_mode=wrapped   BODY_LEAK=True
injection status 'bogus'     body_mode=wrapped   BODY_LEAK=True
injection status ''          body_mode=wrapped   BODY_LEAK=True
```

End-to-end through the real renderer: a bundle item whose envelope carries `injection.status = "Flagged"` (capital F) has its full body — including an "ignore all previous instructions" payload — rendered into the default evidence brief, wrapped in `<<UNTRUSTED-…>>` markers; the lowercase `flagged` baseline is correctly suppressed with `content: omitted`. This violates the #587 acceptance item "Pending, flagged, or error injection state emits metadata only and cannot leak content" and the module's own docstring promise to "fail closed on unknown … injection state". Note the cite/promote/context paths are incidentally safe because they require `status == "clean"` exactly (src/brigade/trust_gate.py:242) — only the brief wrap path fails open.

Fix direction: normalize (`strip().lower()`) at the `injection_status` boundary and fail closed — any status that is not the explicit known-safe `clean` (after pending resolution) should be metadata-only, never wrapped. Add a parametrized negative test over case/whitespace/garbage variants; the current tests only cover exact lowercase `pending`.

### 2. CI `lint` fails — mypy errors in the new module

```
src/brigade/trust_gate.py:405: error: Argument 1 to "matching_transition" has incompatible type "list[dict[str, Any]]"; expected "list[Mapping[str, Any]]"
src/brigade/trust_gate.py:467: error: Argument 1 to "dict" has incompatible type "Any | dict[Any, Any] | None"
src/brigade/trust_gate.py:514: error: Argument 1 to "matching_transition" has incompatible type "list[dict[str, Any]]"; expected "list[Mapping[str, Any]]"
```

Reproduced locally with the repo's pinned mypy. `Sequence` for the `matching_transition` parameter (as mypy suggests) fixes 405/514.

### 3. CI `repo-metadata` fails — stale command inventory

The PR adds `brigade evidence trust review` (src/brigade/cli/evidence.py:65) but does not regenerate `docs/command-inventory.md`. Running `BRIGADE_EXTRAS=1 brigade roadmap commands --write` on the PR head adds the missing row and bumps the evidence count 10 → 11. One command fixes it.

### 4. CI `test (3.11)` fails — two pre-existing tests red on this head

Both reproduced locally:

- `tests/test_work_cmd_work_graph_hardening.py:473` (`test_cycle_via_import_promote_proposed_edges_fails_closed`): the test overwrites `imports[0]["metadata"]` with `{"proposed_edges": …}` (tests/test_work_cmd_work_graph_hardening.py:461), wiping the envelope `_make_import` stamped. The trust gate then correctly fails closed — `error: trust label unknown cannot be promoted` — before the cycle check runs, so the `"cycle" in combined` assertion never matches. The gate ordering is right; the test must preserve `metadata["provenance"]` when injecting `proposed_edges`.
- `tests/test_research_engine.py:336` (`test_adversarial_source_is_wrapped_in_synthesis_and_review`): the adversarial finding ("Ignore all previous instructions") now resolves to flagged/quarantined at `audit_citations`, the citation is `rejected` instead of `accepted`, and the engine enters the single-shot repair — which the scripted backend has no response for (`IndexError: pop from empty list`). The new behavior is arguably the intended #587 enforcement (a flagged finding should not be an accepted citation, and repair is single-shot then `review-rejected`, so no loop), but the test needs a scripted repair response or an updated expectation, and the changelog should state that citing a flagged finding now fails the run after one repair.

The #587 acceptance box "Full `./scripts/verify` passes" is not met with three red CI jobs.

## Should-fix (non-blocking)

1. **Policy inconsistency across item kinds on the same command.** `miseledger trust review` (engines/evidence-ledger/internal/ingest/provenance.go:237, `ReviewTrustLabel`) checks only the digest — no from-label gate. Observed: a `quarantined` engine item was upgraded straight to `reviewed` (and then `verified` via `--to-label`) with nothing but the digest printed by `show --json`. The Python paths explicitly refuse: `review_work_import` / `review_research_finding` raise "cannot review unknown/quarantined content until classification or scan is resolved" (src/brigade/trust_gate.py:461-464, :521-524). So `brigade evidence trust review` rejects quarantined work-imports but happily reviews quarantined miseledger items. Either the engine needs the same from-label gate or the difference needs a documented rationale.
2. **`_upgrade_indexed_verify_receipts` records success before confirming it.** The local `verified` event is appended (src/brigade/receipts_cmd.py:675) before `notify_miseledger_trust`, whose `TrustReviewError` is then swallowed (src/brigade/receipts_cmd.py:684). If the engine notify fails (binary missing, digest drift), the local JSONL claims `verified` while the engine item stays untrusted — and every later `receipts verify` no-ops via `has_matching_transition`, so the divergence can never heal through this path. Notify first, or record the failure instead of swallowing it.
3. **Audit-trail accuracy:** `review_miseledger_item` hardcodes `from_label="untrusted"` (src/brigade/trust_gate.py:635) in the work-events JSONL even when the engine item was `quarantined` — the engine's own `provenance_events` row for the same transition says `quarantined → reviewed`. Two audit records, one transition, different from-labels.
4. **Digest-format inconsistency (trivial, fail-closed):** Python `_require_digest` (src/brigade/trust_gate.py:418) accepts `SHA256:`-prefixed input after lowercasing; Go `ReviewTrustLabel` does `TrimPrefix` before `ToLower` (provenance.go:267), so `SHA256:<hex>` is rejected while `sha256:<hex>` and uppercase bare hex are accepted. Rejecting is safe; align them anyway.

## Verified by probe (acceptance items that hold)

- **Laundering fails closed:** forged top-level `trust_label: "verified"` with no envelope, or combined with an `unknown` envelope, gets no privilege — promote/cite/context deny, brief omits. Label case/whitespace variants (`Verified`, `VERIFIED`, `verified `) all degrade to `unknown` and are excluded.
- **Unknown/missing fail closed:** missing envelope → synthesized `unknown` → blocked from promote/cite/context/brief; missing `hashes.content` → review rejected; tampered text after stamping → digest mismatch → review rejected; forged all-`a` digest rejected on both the Python and engine paths.
- **Idempotency:** work-import review twice → one event, second is `noop`; `brigade evidence trust review miseledger:item:…` duplicate → `noop`, one event; direct `miseledger trust review` three times → one engine event (`TransitionTrustLabel` early-returns when from==to, provenance.go:201-203); second `receipts verify` is a `noop` (test-covered).
- **Caps and pending resolution:** 2-item and 50%-byte caps tested at both boundaries; exact-lowercase `pending` resolves synchronously and a hit becomes metadata-only — confirmed in the rendered brief.
- **Engine gates:** `go vet`/`go test` clean; forged digest rejected rc=1; `trust review` requires the exact current envelope digest and the recomputed text hash.

## Summary

The architecture is right and most of the #587 acceptance list probes clean, but this cannot land with (a) a confirmed content-leak bypass for non-exact injection statuses in the brief wrap path, and (b) three red CI jobs including the full-test gate the issue explicitly requires. Fix the injection-status normalization (fail closed on anything not exactly `clean`), fix the mypy errors, regenerate the command inventory, and update the two red tests — the should-fix items can land in the same push or a fast follow-up.

**Comment by solomonneas** [comment]
VERDICT: READY TO LAND

Independent re-verification of #965 at head `ffde8dbb` against the 22:52:12Z NEEDS WORK review. Executed in a fresh scratch clone (`/tmp/pr965-verify`, deleted after), Python 3.12.3 venv with `pip install -e ".[dev]"`. All four blocking findings are fixed; every check below was executed, not just read.

## 1. Injection-status probe table (blocking finding) — FIXED

Drove the real renderer (`evidence_brief.render_evidence_bundle`) end-to-end with a bundle item whose envelope carries each status variant and whose body is a prompt-injection payload ("Ignore all previous instructions…" plus a unique marker), cross-checked against `trust_gate.admit_consumer` body_mode:

```
injection status     body_mode  BODY_LEAK
'flagged'            metadata   False
'Flagged'            metadata   False
'FLAGGED'            metadata   False
'flagged '           metadata   False
'Error'              metadata   False
'bogus'              metadata   False
''                   metadata   False
'clean'              wrapped    True   (explicit known-safe; wraps as intended)
'Clean'              wrapped    True
'CLEAN'              wrapped    True
'clean '             wrapped    True
```

All seven attack variants are metadata-only — the rendered default brief shows `content: omitted` and the payload marker is absent. Only normalized-`clean` wraps, matching the fail-closed fix direction from the review. Confirmed in source: `provenance.injection_status` now `strip().lower()`s (src/brigade/provenance.py:485), `injection_blocks_content` is fail-closed on anything != `clean` (src/brigade/trust_gate.py:163-167), and the wrap branch requires `status == "clean"` (src/brigade/trust_gate.py:245-246). The new parametrized negatives (`test_evidence_brief_injection_status_variants_are_metadata_only`, `test_admit_consumer_non_clean_injection_variants_are_metadata_only`, `test_admit_consumer_normalized_clean_may_wrap`) pass 20/20.

## 2. mypy — FIXED

`mypy src/brigade/trust_gate.py` → `Success: no issues found in 1 source file`. The three errors at 405/467/514 are gone; CI `lint` is green on this head.

## 3. Command inventory — FIXED

`BRIGADE_EXTRAS=1 brigade roadmap commands --check` → rc=0, `inventory: docs/command-inventory.md (current)`, `[ok] brigade evidence: 11 CLI path(s)`. CI `repo-metadata` is green.

## 4. The two previously-failing `test (3.11)` tests — FIXED

```
pytest -q tests/test_work_cmd_work_graph_hardening.py::test_cycle_via_import_promote_proposed_edges_fails_closed \
          tests/test_research_engine.py::test_adversarial_source_is_wrapped_in_synthesis_and_review
# 2 passed
```

Both reproduced-red on the old head, both green now. The CI `test (3.11)` job on `ffde8dbb` also completed **success** (run 32078435622, 17m5s); `test (3.10)` and `test (3.12)` were still queued at comment time, but 3.11 — the previously red job — is green, alongside `lint` and `repo-metadata`.

## 5. Touched-module suites

```
pytest -q tests/test_provenance_envelope.py tests/test_aboyeur.py tests/test_context_cmd.py \
          tests/test_receipts_cmd.py tests/test_evidence_cmd.py tests/test_work_cmd_imports.py \
          tests/test_research_provenance.py tests/test_work_cmd_work_graph_hardening.py \
          tests/test_research_engine.py
# 636 passed in 55.34s
```

Matches the fixer's claim of 636 passed. The delta `2c336b20..ffde8dbb` is tightly scoped — 8 files: the normalization/fail-closed fix, the mypy typing fixes, the inventory regen, the two test updates, the new parametrized negatives, and the changelog note. No scope creep, no secrets. The four should-fix items (engine from-label gate, receipts-verify notify ordering, miseledger from-label audit, digest-format alignment) remain deferred to a follow-up, as the review allowed.

**Comment by solomonneas** [comment]
Holding this merge. CI run `32078435622` is green on `ffde8dbb`, but four review threads are open and unanswered, and three of them describe defects in the trust-enforcement surface this PR exists to add. I checked them against the diff rather than going on the comments alone.

When I first looked at this PR the unresolved count was **0**; these landed afterward, so the READY TO LAND re-review predates them.

## Blocker 1: quarantine can be downgraded to untrusted

`src/brigade/trust_gate.py:212` — `resolve_pending_injection` gates only on the injection status:

```python
if status != "pending":
    return updated, status or "error"
```

Nothing consults the item's *label*. An item labeled `quarantined` whose injection status is set to `pending` therefore takes the clean-scan path and is released to `untrusted`, which is content-eligible for promote, context, and cite. That is a bypass of quarantine, reachable by setting a single field.

The docstring says "A clean scan is a consumer-local release to `untrusted`", which is the right intent for an item that was never quarantined. It needs to refuse the release when the label is already `quarantined`.

## Blocker 2: `context` entitlement appears inverted

`src/brigade/trust_gate.py:251` — the hardcoded fallback in `admit_consumer` is:

```python
allowed=entitlement in {"brief", "brief_wrapped", "show", "search"}
```

`context` is absent. Per the reviewer, `src/brigade/fixtures/trust-policy.v1.json` grants `context` to no label, while `untrusted` is explicitly allowed for `context` elsewhere. The net effect is that `reviewed` and `verified` items are denied a surface that `untrusted` items are admitted to, which inverts the trust ordering this PR is establishing.

Worth confirming against the fixture and either granting `context` to the higher labels or documenting deliberately why `context` is untrusted-only.

## Blocker 3: verified status reported when the trust transition failed

`src/brigade/receipts_cmd.py:686` — `_upgrade_indexed_verify_receipts` swallows `notify_miseledger_trust()` failures, then appends a local `verified` event and reports `status="verified"`. An operator reading that output concludes the item was verified in MiseLedger when the transition actually failed. A false attestation is worse than a reported failure, particularly on the surface that #584 to #587 exists to make trustworthy.

## Non-blocking: `_append_jsonl` rewrites the whole file

`src/brigade/trust_gate.py:356` reads the entire JSONL and rewrites it on every append. That is O(n) per event, and two concurrent appenders lose events to last-writer-wins. Other JSONL helpers in this repo use append mode. For an audit trail the durability point matters more than the latency one, but it is separable from the three above.

## Why I am not resolving these

All four describe behavior this PR introduces, in its own security surface. The rule I have applied across this campaign is that a thread gets resolved only when it describes pre-existing behavior the PR strictly improves and the residue is tracked in an issue; when a thread has described a defect the PR itself introduced, holding has been correct every time — #944's reservation-identity scoping, #951's inherited `started_at`, and #960's uppercase-digest and forensic fail-open bypasses were all real and all subsequently fixed.

Green CI is not evidence against any of these. None of them fail a test: the quarantine downgrade and the entitlement inversion are policy logic, and the swallowed notify failure is deliberately caught.

The fixes look small — a label check before release, a fixture or fallback correction, and propagating the notify error into the reported status. Happy to land as soon as they are in.

**Comment by solomonneas** [comment]
VERDICT: READY TO LAND

Independent re-verification of #965 at head `62ff78d5` against the 23:26:18Z shepherd hold (four late review threads). Executed in a fresh scratch clone (`/tmp/pr965-reverify`, deleted after), Python venv with `pip install -e ".[dev]"`. Read the shepherd comment, all four Copilot threads, and the fixer summary; every check below was executed, not just read. Reviewer: Brigade worker k3_kimi (Kimi Code CLI).

## Probe 1: quarantine downgrade via injection-status-only gating — FIXED

Built an item labeled `quarantined` with injection status `pending` and a benign body, drove `resolve_pending_injection` + `admit_consumer` directly:

- Scan resolves `clean`, label **stays `quarantined`** (no release to `untrusted`).
- `admit_consumer` denies all 7 surfaces (`brief`, `brief_wrapped`, `show`, `search`, `cite`, `promote`, `context`) with `body_mode=omit`; `citation_allowed`/`context_allowed` are `False`; `promotion_blocker` = "trust label quarantined cannot be promoted".
- Case variant `Pending` is also denied; control (`untrusted`+`pending`, clean scan) is still admitted to `context`, so the fix doesn't over-block.

## Probe 2: inverted context entitlement — FIXED

Built the full label × surface admission matrix (clean injection) and asserted monotonicity:

```
unknown     -> {}
untrusted   -> {brief, brief_wrapped, cite, context, promote, search, show}
reviewed    -> {brief, brief_wrapped, cite, context, promote, search, show}
verified    -> {brief, brief_wrapped, cite, context, promote, search, show}
quarantined -> {}
```

Monotonic `unknown ⊆ untrusted ⊆ reviewed ⊆ verified` holds; `reviewed` and `verified` reach `context`. No higher-trust label reaches fewer surfaces than a lower one.

## Probe 3: false verified attestation on swallowed notify failure — FIXED

Forced `notify_miseledger_trust` to raise `TrustReviewError` through the real `receipts_cmd.verify_payload` flow: reported status is `verify-pending` (with the error), **not** `verified`; no local verified event is appended; a retry still reports `verify-pending` (healable instead of a false noop).

## Probe 4: earlier case-variant brief probes — STILL HOLD

Drove the real renderer (`evidence_brief`) end-to-end with a prompt-injection payload body across 14 status variants. All 10 non-clean variants (`flagged`, `Flagged`, `FLAGGED`, `flagged `, `Error`, `ERROR`, `bogus`, `ok`, `''`, `'  '`) are metadata-only (`content: omitted`, no payload marker, no `<<UNTRUSTED-` wrap); only normalized-clean (`clean`, `Clean`, `CLEAN`, `clean `) wraps.

## Non-blocking thread: `_append_jsonl` — FIXED

Confirmed append mode + `fsync` in source (src/brigade/trust_gate.py:356); three sequential appends all survive.

## Suites and gates

```
python -m pytest -q tests/test_provenance_envelope.py tests/test_aboyeur.py \
  tests/test_context_cmd.py tests/test_receipts_cmd.py tests/test_evidence_cmd.py \
  tests/test_work_cmd_imports.py tests/test_research_provenance.py \
  tests/test_work_cmd_work_graph_hardening.py tests/test_research_engine.py
# 640 passed in 55.76s

mypy src/brigade/trust_gate.py src/brigade/provenance.py src/brigade/receipts_cmd.py
# Success: no issues found in 3 source files

BRIGADE_EXTRAS=1 brigade roadmap commands --check
# rc=0; inventory: docs/command-inventory.md (current); [ok] brigade evidence: 11 CLI path(s)
```

All three shepherd-verified defects are fixed with correct fail-closed semantics, the non-blocking durability item is fixed too, and the earlier READY findings remain green. No code edited, not merging, not marking ready — per the hold protocol the threads stay for the shepherd to resolve.

File: src/brigade/receipts_cmd.py [hunk 1]
```diff
@@ -582,11 +583,121 @@ def _summary(items: list[dict[str, Any]]) -> dict[str, int]:
...
+            trust_gate.notify_miseledger_trust(
+                target,
+                external_id,
+                content_hash,
+                to_label="verified",
+                operator_command=trust_gate.OPERATOR_VERIFY,
+            )
+        except trust_gate.TrustReviewError:
+            pass
+        upgrades.append({"status": "verified", "item_ref": item_ref, "event": event})
```
[the remark this answers is not in the document]
Comment (solomonneas) [reply src/brigade/receipts_cmd.py]: Verified fixed. A failed trust transition no longer reports success:

```python
except trust_gate.TrustReviewError as exc:
    upgrades.append({
        "status": "verify-pending",
        "item_ref": item_ref,
        "envelope_content_hash": content_hash,
        "error": str(exc),
    })
```

`status` is now `verify-pending` carrying the error text, instead of appending a local `verified` event and reporting `verified`. An operator reading this output can no longer conclude the item was verified in MiseLedger when the notify actually failed. That false-attestation risk was the whole of my concern. Resolving.

File: src/brigade/trust_gate.py [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+    return updated
+
+
+def _append_jsonl(path: Path, payload: Mapping[str, Any]) -> None:
+    line = json.dumps(payload, sort_keys=True, separators=(",", ":"), ensure_ascii=False) + "\n"
+    existing = ""
+    if path.is_file():
+        existing = path.read_text(encoding="utf-8")
+    path.parent.mkdir(parents=True, exist_ok=True)
+    localio.write_text_atomic(path, existing + line)
```
[the remark this answers is not in the document]
Comment (solomonneas) [reply src/brigade/trust_gate.py]: Verified fixed, and beyond what the thread asked for.

`_append_jsonl` now opens in append mode and syncs:

```python
with path.open("a", encoding="utf-8") as handle:
    handle.write(line)
    handle.flush()
    os.fsync(handle.fileno())
```

That removes the read-entire-file-and-rewrite pattern, so appends are O(1) rather than O(n) and concurrent appenders no longer lose events to last-writer-wins. The added `fsync` also gives durability across a crash, which the original comment did not ask for but matters for an audit trail. Resolving.

File: src/brigade/trust_gate.py [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+        )
+    if entitlement == "brief" and allows(label, "brief") and status == "clean":
+        return ConsumerAdmission(label=label, allowed=True, body_mode="full", injection_status=status, envelope=env)
+    if entitlement in {"brief", "brief_wrapped"} and allows(label, "brief_wrapped") and status == "clean":
+        return ConsumerAdmission(label=label, allowed=True, body_mode="wrapped", injection_status=status, envelope=env)
+    if allows(label, entitlement):
+        return ConsumerAdmission(label=label, allowed=True, body_mode="full", injection_status=status, envelope=env)
+    if entitlement in {"cite", "promote", "context"} and label == "untrusted" and status == "clean":
+        # Default work-import and research paths stay usable after a clean scan.
+        # Unknown and quarantined remain excluded above.
```
[the remark this answers is not in the document]
Comment (solomonneas) [reply src/brigade/trust_gate.py:257]: Verified fixed, and the design is better than what I suggested.

I proposed a label guard inside `resolve_pending_injection`. The fix instead enforces at the single admission chokepoint, which is stronger. `resolve_pending_injection` only ever *sets* `quarantined` (`if new_status == "flagged": trust["label"] = "quarantined"`) and never clears it, so a quarantined label survives injection resolution. `admit_consumer` then derives the label *after* that call and hard-denies:

```python
if label in {"unknown", "quarantined"}:
    return ConsumerAdmission(label=label, allowed=False, body_mode="omit", ...)
```

`allowed=False` with `body_mode="omit"` closes the promote/context/cite path I was worried about, and `test_admit_consumer_rejects_laundered_unknown_and_quarantined_claims` pins the anti-laundering property directly. Enforcing at one gate rather than scattering guards through the resolution path is the right call. Resolving.
